src/socket-activate/socket-activate.c

```diff
@@ -97,24 +98,40 @@ static int open_sockets(int *ret_epoll_fd) {
                 count++;
         }
 
+        if (count > 1 && !arg_accept && arg_inetd)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                       "--inetd only supported with a single file descriptor, "
+                                       "or with --accept.");
+
+        size_t n_fdnames = strv_length(arg_fdnames);
+        if (arg_fdnames && n_fdnames != (size_t) count)
+                log_warning("The number of fd names is different than number of fds: %zu vs %i",
+                            n_fdnames,
+                            count);
```

Comment:
this fails to account for --accept case, where we listen on multiple fds but only pass one accept()-ed connection fd to child, which means there shall be exactly one fdname but possibly with more than one fd.

dunno, I think this check can be left as-is in exec_process. It's just a warning after all, and we lose nothing even if it's checked a bit late.